recipes/MEDIA/ASR/seq2seq/train.py

```diff
@@ -0,0 +1,360 @@
+        start = int(float(start_seg) * hparams["sample_rate"])
+        stop = int(float(end_seg) * hparams["sample_rate"])
+        speech_segment = {"file": wav, "start": start, "stop": stop}
+        sig = sb.dataio.dataio.read_audio(speech_segment)
+        return sig
+
+    sb.dataio.dataset.add_dynamic_item(datasets, audio_pipeline)
+
+    # 3. Define text pipeline:
+    @sb.utils.data_pipeline.takes("char")
+    @sb.utils.data_pipeline.provides("char_list", "char_encoded")
+    def text_pipeline(char):
+        char_list = char.strip().split()
+        yield char_list
+        char_encoded = label_encoder.encode_sequence_torch(char_list)
+        yield char_encoded
+
+    sb.dataio.dataset.add_dynamic_item(datasets, text_pipeline)
+
+    # 4. Create a label encoder instead of a tokenizer for our tag list:
+    lab_enc_file = hparams["save_folder"] + "/labelencoder.txt"
+    label_encoder.load_or_create(
+        path=lab_enc_file,
+        from_didatasets=[train_data],
+        output_key="char_list",
+        special_labels={"blank_label": hparams["blank_index"]},
+        sequence_input=True,
+    )
+
+    # 5. Set output:
+    sb.dataio.dataset.set_output_keys(
+        datasets, ["id", "sig", "char_encoded"],
+    )
+
+    # 6. Make DataLoaders and shuffle if needed:
+    dataloader_train = torch.utils.data.DataLoader(
+        train_data,
+        batch_size=hparams["batch_size"],
+        num_workers=hparams["dataloader_options"]["num_workers"],
+        collate_fn=PaddedBatch,
+        shuffle=hparams["dataloader_options"]["shuffle"],
+    )
+    dataloader_valid = torch.utils.data.DataLoader(
+        valid_data,
+        batch_size=hparams["batch_size"],
+        num_workers=hparams["dataloader_options"]["num_workers"],
+        collate_fn=PaddedBatch,
+        shuffle=hparams["dataloader_options"]["shuffle"],
+    )
+    dataloader_test = torch.utils.data.DataLoader(
+        test_data,
+        batch_size=hparams["test_batch_size"],
+        num_workers=hparams["test_dataloader_options"]["num_workers"],
+        collate_fn=PaddedBatch,
+    )
+
+    return dataloader_train, dataloader_valid, dataloader_test, label_encoder
+
+
+if __name__ == "__main__":
+
+    # Load hyperparameters file with command-line overrides.
+    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
+    with open(hparams_file) as fin:
+        hparams = load_hyperpyyaml(fin, overrides)
+
+    # If distributed_launch=True then
+    # create ddp_group with the right communication protocol.
+    sb.utils.distributed.ddp_init_group(run_opts)
+
+    # Create experiment directory.
+    sb.create_experiment_directory(
+        experiment_directory=hparams["output_folder"],
+        hyperparams_to_save=hparams_file,
+        overrides=overrides,
+    )
+
+    # Due to DDP, we do the preparation ONLY on the main python process
+    run_on_main(
+        prepare_media,
+        kwargs={
+            "data_folder": hparams["original_data_folder"],
```

Comment:
Ok, there's a naming issue here. Original_data_folder == data_folder ? If so, you will need to alter the YAML according to that.

Reply:
mistake thx